python/src/iceberg/catalog/base.py

```diff
@@ -0,0 +1,223 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+from abc import ABC, abstractmethod
+from typing import Dict, List, Optional, Tuple
+
+from iceberg.schema import Schema
+from iceberg.table.base import PartitionSpec, Table
+
+
+class Catalog(ABC):
+    """Base Catalog for table operations like - create, drop, load, list and others.
+
+    Attributes:
+        name(str): Name of the catalog
+        properties(Dict[str, str]): Catalog properties
+    """
+
+    def __init__(self, name: str, properties: Dict[str, str]):
+        self._name = name
+        self._properties = properties
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def properties(self) -> Dict[str, str]:
+        return self._properties
+
+    @abstractmethod
+    def create_table(
+        self,
+        *,
```

Comment:
What is your rationale for this? I'd probably just allow whatever the caller prefers.

Reply:
+1, I can't think of a reason why we would want to stop someone from doing `create_table(("foodb"), "footable", schema)` and force them to do `create_table(namespace=("foodb"), name="footable", schema=schema)`. Both seem valid to me.

Reply:
Yeah, I think we should allow callers to choose. You make a good point about internal APIs, where this is probably more relevant. But forcing style on library callers seems unnecessary to me.

Reply:
Makes sense. Will remove the enforcement.

Reply:
Fixed in a recent commit. Marking it resolved.